test/mitmproxy/console/test_flowlist.py

```diff
@@ -0,0 +1,31 @@
+import mitmproxy.tools.console.flowlist as flowlist
+from mitmproxy.tools import console
+from mitmproxy import proxy
+from mitmproxy import options
+from .. import tservers
+import pytest
+from unittest import mock
+
+
+class UrlError(Exception):
+    pass
+
+
+def mock_log(message):
+    if "Invalid URL" in message:
+        raise UrlError(message)
+
+
+class TestFlowlist(tservers.MasterTest):
+    def mkmaster(self, **opts):
+        if "verbosity" not in opts:
+            opts["verbosity"] = 1
+        o = options.Options(**opts)
+        return console.master.ConsoleMaster(o, proxy.DummyServer())
+
+    @mock.patch('mitmproxy.tools.console.signals.status_message.send', side_effect = mock_log)
```

Comment:
It is probably easier if you directly check that `signals.status_message.send` was called with specific arguments, instead of using a side effect + exception wrapper.
E.g., `Mock.assert_called_once_with` or similar.